### PR TITLE
Update E2E Readme and simplify configuration

### PR DESCRIPTION
**Pull request by chatton**
## Description

This PR modifies the E2E readme to make it up to date with the ability for E2Es to run full suites at a time.

I decided to strip out a lot of the README which I think was not providing value.

 I removed the code samples (there are enough tests to provide code samples at this point) and unnecessary detail about the tool that generates the gh matricies.

This PR also streamlines the default config file to be tiny and populate most things via defaults. The existing file was renamed to `sample.config.extended.yaml` to be used as a reference with all the fields set.

closes: #6646

---

Before we can merge this PR, please make sure that all the following items have been
checked off. If any of the checklist items are not applicable, please leave them but
write a little note why.

- [ ] Targeted PR against the correct branch (see [CONTRIBUTING.md](https://github.com/cosmos/ibc-go/blob/main/docs/dev/pull-requests.md#pull-request-targeting)).
- [ ] Linked to GitHub issue with discussion and accepted design, OR link to spec that describes this work.
- [ ] Code follows the [module structure standards](https://github.com/cosmos/cosmos-sdk/blob/main/docs/build/building-modules/11-structure.md) and [Go style guide](../docs/dev/go-style-guide.md).
- [ ] Wrote unit and integration [tests](https://github.com/cosmos/ibc-go/blob/main/testing/README.md#ibc-testing-package).
- [ ] Updated relevant documentation (`docs/`).
- [ ] Added relevant `godoc` [comments](https://blog.golang.org/godoc-documenting-go-code).
- [ ] Provide a [conventional commit message](https://github.com/cosmos/ibc-go/blob/main/docs/dev/pull-requests.md#commit-messages) to follow the repository standards.
- [ ] Include a descriptive changelog entry when appropriate. This may be left to the discretion of the PR reviewers. (e.g. chores should be omitted from changelog)
- [ ] Re-reviewed `Files changed` in the GitHub PR explorer.
- [ ] Review `SonarCloud Report` in the comment section below once CI passes.

### CODE REVIEW

File: e2e/README.md [hunk 1]
```diff
@@ -1,17 +1,8 @@
...
-   - b. [Running the tests with custom images](#running-tests-with-custom-images)
-   - b. [Code samples](#code-samples)
-     - [Setup](#setup)
-     - [Creating test users](#creating-test-users)
-     - [Waiting](#waiting)
-     - [Query wallet balances](#query-wallet-balances)
-     - [Broadcasting messages](#broadcasting-messages)
-     - [Starting the relayer](#starting-the-relayer)
-     - [Arbitrary commands](#arbitrary-commands)
-     - [IBC transfer](#ibc-transfer)
```
Comment (chatton) [inline e2e/README.md:14]: removed these in favour of just having more tests that have code samples!

File: e2e/sample.config.yaml [hunk 2]
```diff
@@ -1,49 +1,18 @@
...
-# Required only for upgrade tests.
-# Chain A will be upgraded the specified tag.
-# The plan name must be registered as an upgrade handler in the given tag.
-upgrade:
-  planName: ""
-  tag: ""
+---
+chains:
+  - tag: main # override with CHAIN_A_TAG
+  - tag: main # override with CHAIN_B_TAG
```
Comment (chatton) [inline e2e/sample.config.yaml:18]: technically this is all you actually need now to have a valid config file assuming you want to use all the defaults (hermes 1.10.0, simd etc. )

File: e2e/testsuite/testconfig.go [hunk 3]
```diff
@@ -341,7 +341,44 @@ func fromFile() (TestConfig, bool) {
...
+			getDefaultHermesRelayerConfig(),
+			getDefaultHyperspaceRelayerConfig(),
+		}
+	}
+
+	if tc.CometBFTConfig.LogLevel == "" {
+		tc.CometBFTConfig.LogLevel = "info"
+	}
+
+	return tc
```
Comment (chatton) [inline e2e/testsuite/testconfig.go:381]: enables the minimal config, Happy to move this to a separate PR if people would prefer.

File: e2e/README.md [hunk 4]
```diff
@@ -42,187 +33,99 @@ Tests can be run using a Makefile target under the e2e directory. `e2e/Makefile`
...
+pushed [here](https://github.com/orgs/cosmos/packages?repo_name=ibc-go).
 
-### Example Command
+### Example of running a single test
 
-```sh
-export CHAIN_IMAGE="ghcr.io/cosmos/ibc-go-simd"
-export CHAIN_A_TAG="main"
-export CHAIN_BINARY="simd"
+> NOTE: environment variables can be set to override a config file, but the config file can still be used to set
```
Comment (gjermundgaraba) [inline e2e/README.md:63]: Very minor nit: Maybe something like this to make it even clearer?

```suggestion
> NOTE: environment variables can be set to override one or more config file variables, but the config file can still be used to set
```

File: e2e/README.md [hunk 5]
```diff
@@ -42,187 +33,99 @@ Tests can be run using a Makefile target under the e2e directory. `e2e/Makefile`
...
-
-Every standard test will start with this. This creates two chains and a relayer,
-initializes relayer accounts on both chains, establishes a connection and a channel
-between the chains.
-
-Both chains have started, but the relayer is not yet started.
+### Example of running a full testsuite
 
-The relayer should be started as part of the test if required. See [Starting the Relayer](#starting-the-relayer)
+> NOTE: not all tests may support full parallel runs due to possible chain wide modifications such as params / gov
```
Comment (gjermundgaraba) [inline e2e/README.md:90]: Maybe link down to the "When to use t.Parallel" section?

File: e2e/README.md [hunk 6]
```diff
@@ -42,187 +33,99 @@ Tests can be run using a Makefile target under the e2e directory. `e2e/Makefile`
...
-
-```go
-t.Run("start relayer", func(t *testing.T) {
-  s.StartRelayer(relayer)
-})
-```
+In order to run tests outside the context of the Makefile (e.g. from an IDE)
 
-### Arbitrary commands
+The main thing that needs to be set is the `E2E_CONFIG_PATH` environment variable. This should point to a valid config
```
Comment (gjermundgaraba) [inline e2e/README.md:104]: This is not entirely true. You don't have to set it, the default location works (this is what I currently use).

File: e2e/README.md [hunk 7]
```diff
@@ -42,187 +33,99 @@ Tests can be run using a Makefile target under the e2e directory. `e2e/Makefile`
...
-t.Run("send IBC transfer", func(t *testing.T) {
-  chainATx, err = chainA.SendIBCTransfer(ctx, channelA.ChannelID, chainAWallet.KeyName, walletAmount, nil)
-  s.Require().NoError(err)
-  s.Require().NoError(chainATx.Validate(), "chain-a ibc transfer tx is invalid")
-})
-```
+### Test Suites
 
-Broadcast a `MsgTransfer`.
+In order for tests to be run in parallel, we create the chains in `SetupSuite`, and each test is in charge of
```
Comment (gjermundgaraba) [inline e2e/README.md:116]: Would be nice with some kind of example or pointing to an example

**Review by gjermundgaraba — COMMENTED** [review]
Some nits, but overall this is a huge improvement! Really great work 🙏

**Review by gjermundgaraba — APPROVED** [review]

**Comment by chatton** [comment]
Sorry lost track of this PR when I was out, will get back on this ASAP! Thanks for the reviews @gjermundgaraba  / @DimitrisJim  great suggestions 🥇

File: e2e/README.md (hunk 6 above)
Comment (chatton) [reply e2e/README.md:104]: ah good point, will just make a note about this env being what you need to change.
